prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,1127 @@
+            // that may or may not have happened.
+            checkActionLogTable(expectedButStrangeObservationFilterEntry)
+        }
+    }
+
+    @Test
+    fun `should send valid FHIR report filtered by condition code 94558-5`() {
+        // set up
+        val reportContents =
+            listOf(
+                File(MULTIPLE_OBSERVATIONS_FHIR_URL).readText()
+            ).joinToString()
+
+        val reportPair = createReportsWithLineage(reportContents)
+        val receiveReport = reportPair.first
+        val convertReport = reportPair.second
+        val queueMessage = generateQueueMessage(convertReport, reportContents, fhirSenderWithNoTransform)
+        val fhirFunctions = createFHIRFunctionsInstance()
+
+        // make sure action table has only what we put in there
+        checkActionTable(listOf(TaskAction.convert, TaskAction.receive))
+
+        // execute
+        var receiverSetupData = listOf(
+            ReceiverSetupData("x", conditionFilter = observationFilter),
+            ReceiverSetupData("y"),
+        )
+        var receivers = createReceivers(receiverSetupData)
+        val org = createOrganizationWithReceivers(receivers)
+        val fhirRouter = createFHIRRouter(org)
+        fhirFunctions.doRoute(queueMessage, 1, fhirRouter)
+
+        // check results
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            // did the report get pushed to blob store correctly and intact?
+            val routedReports = verifyLineageAndFetchCreatedReportFiles(convertReport, receiveReport, txn, 2)
+            val reportAndBundles =
+                routedReports.map {
+                    Pair(
+                        it,
+                        BlobAccess.downloadBlobAsByteArray(it.bodyUrl, getBlobContainerMetadata())
+                    )
+                }
+
+            var fhirBundlesAsObjectsOnly = reportAndBundles.map { it.second.toString(Charsets.UTF_8) }
+                                                           .map { FhirTranscoder.decode(it) }
+
+            var fhirBundleReceiverX = fhirBundlesAsObjectsOnly[0]
+            var fhirBundleReceiverY = fhirBundlesAsObjectsOnly[1]
+
+            // there should only be one observation of five remaining, and the code of that observation
+            // should be 94558-5
+            assertEquals(1, fhirBundleReceiverX.getObservations().size)
+            assertEquals(1, fhirBundleReceiverX.getObservations()[0].code.coding.size)
+            assertThat(fhirBundleReceiverX.getObservations()[0].code.coding[0].code.equals("94558-5"))
+
+            // for receiver Y all five observations should be intact
+            assertEquals(5, fhirBundleReceiverY.getObservations().size)
+            val expectedCodes = listOf("94558-5", "95418-0", "95417-2", "95421-4", "95419-8")
+            for (i in 0..fhirBundleReceiverX.getObservations().size - 1) {
```

Comment:
Intellij doesn't like this syntax

Reply:
what do you mean it doesn't like this syntax? can you be more specific? everything works AFAIK

Reply:
![Screenshot 2024-05-31 at 9 31 52 AM](https://github.com/CDCgov/prime-reportstream/assets/118766341/c49d8550-f9d8-4cb6-8a38-8e086a0b85e1)

Reply:
✅